prompt2model/dataset_generator/readme.md

```diff
@@ -5,26 +5,21 @@
 - `DatasetGenerator`: An abstract class to generate datasets.
 - `DatasetSplit`: An enumeration class defining dataset types (`TRAIN`,
 `VALIDATION`, `TEST`).
-- `OpenAIDatasetGenerator`: A concrete class
+- `PromptBasedDatasetGenerator`: A concrete class
 for dataset generation using GPT-3.5 API.
 
 ## Getting Started
 
 - **Import the Modules**:
 
 ```python
-from prompt2model.dataset_generator import OpenAIDatasetGenerator, DatasetSplit
-from prompt2model.prompt_parser import OpenAIInstructionParser, TaskType
+from prompt2model.dataset_generator import PromptBasedDatasetGenerator, DatasetSplit
+from prompt2model.prompt_parser import PromptBasedInstructionParser, TaskType
 ```
 
-- **Setup OpenAI API Key**:
+- **Setup API Key**:
 
-```python
-api_key = "<your-api-key>"
-dataset_generator = OpenAIDatasetGenerator(api_key)
-```
-
-Or, set as an environment variable:
+Set an API key as an environment variable:
 
 ```bash
 export OPENAI_API_KEY="<your-api-key>"
```

Comment:
This should not be `OPENAI_API_KEY`.

Reply:
It should be if you're using OpenAI, so I modified the doc a little.